docs/src/pages/index.mdx

```diff
@@ -0,0 +1,178 @@
+---
+layout: ../layouts/wiki.astro
+---
+
+import InstallCommands from "@/components/install-commands.astro";
+
+# Remeda
+
+Fully written in TypeScript for powerful, type-safe, and functional utilities.
+
+<InstallCommands />
+
+## 🧰 Expansive
+
+Remeda offers a wide range of common everyday utility functions so that you
+don't have to deal with the minutiae of writing them yourself. We ensure that
+our algorithms are efficient, optimized, and cover all edge cases properly. We
+offer a wide range of functions, from simple utilities like
+[`partition`](/docs#partition) and building objects with
+[`pullObject`](/docs#pullObject), to _complex_ algorithms like
+[`takeFirstBy`](/docs#takeFirstBy).
+
+```ts
+type User = {
+  firstName: string;
+  lastName: string;
+  lastLogin: number;
+};
+
+decalre const USERS: User[];
+
+R.sortBy(
+  USERS,
+  [R.prop("lastLogin"), "desc"],
+  R.prop("lastName"),
+  R.prop("firstName"),
+);
+```
+
+## 👷 Fully Typed
+
+Libraries like Lodash and Ramda are written in JavaScript, with types added
+later. Remeda, on the other hand, is fully written in TypeScript! This means
+types are deeply integrated into the code, allowing us to shift the focus from
+runtime errors to compile-time errors, so you catch errors in your IDE and CI,
+not in your logs, **and not in production**.
+
+We strive to provide the best possible real-life types for all functions, often
+going beyond what is offered by the TypeScript library, so you don't need to use
+other libraries like 'type-fest' and 'ts-reset' just to get the types you
+expected.
+
+```ts
+const DATA = [1, 2, 3] as const;
+
+const builtin = DATA.map(DATA, (x) => x.toString());
+//    ^? string[]
+
+const withRemeda = R.map(DATA, (x) => x.toString());
+//    ^? [string, string, string]
+```
+
+## 🦾 Functional
+
+Functional programming is beneficial, making code more readable. However, there
+are situations where you don't need "pipes" and want to call just a single
+function. In Remeda, **all** functions have automatic currying built-in. This
+means the same imported function can be called either regularly (what we call
+_dataFirst_) or with the first parameter, _data_, omitted (what we call
+_dataLast_). When a dataLast invocation is detected, Remeda creates a curried
```

Comment:
**question:** do we want to switch to using "dataFirst" and "dataLast" everywhere?

so far convention has been to use "data-first" and "data-last", which is standard, see e.g. [this article](https://www.javierchavarri.com/data-first-and-data-last-a-comparison/) or [lodash wiki](https://github.com/lodash/lodash/wiki/FP-Guide). i think this is grammatically correct, as data-first is a [compound modifier](https://en.wikipedia.org/wiki/Compound_modifier)

whatever we decide to use though, we should be consistent—we can edit the readme after this if we decide to prefer dataFirst and dataLast

Reply:
…i just now realize we use "data first" (with spaces) in the documentation…

Reply:
The tag was renamed from "data-first" to "dataFirst" because typedoc doesn't support it, and once I did that I got used to calling it like that, but I think it's fine to change it to what is the most common way.